api/mesh/v1alpha1/dataplane_insight_helpers.go

```diff
@@ -56,13 +56,17 @@ func (x *DataplaneInsight) IsOnline() bool {
 	return false
 }
 
-func (x *DataplaneInsight) GetSubscription(id string) (int, *DiscoverySubscription) {
-	for i, s := range x.GetSubscriptions() {
+func (x *DataplaneInsight) AllSubscriptions() []generic.Subscription {
+	return generic.AllSubscriptions[*DiscoverySubscription](x)
+}
+
+func (x *DataplaneInsight) GetSubscription(id string) generic.Subscription {
+	for _, s := range x.GetSubscriptions() {
```

Comment:
Unlike `UpdateSubscription` this one also should be pretty straightforward to convert into generic method like `AllSubscriptions`

Reply:
like the commit I added?

Reply:
yes 👍